docs/WASI-design-principles.md

```diff
@@ -0,0 +1,95 @@
+# WASI Design Principles
+
+## Capability-based security
+
+WASI is built using capability-based security principles. Access to external resources is always represented by *handles*, which are special values that are *unforgeable*, meaning there's no way to coerce an arbitrary integer or other type of value into a handle. WASI is also aiming to have no *ambient authorities*, meaning that there should be no way to request a handle purely by providing a string or other user-controlled identifier providing the name of a resource. With these two properties, the only ways to obtain access to resources are to be explicitly given handles, or to perform operations on handles which return new handles.
```

Comment:
Would you mind wrapping these long paragraphs at 80 chars?

Reply:
Done.